--- tokenizer.sh	2022-01-21 13:26:10.150335638 +0100
+++ pintokenizer.sh	2022-01-21 13:25:37.229795176 +0100
@@ -33,3 +33,11 @@
 #--[ User output ]------------------------------------------------------------------------------
 
 echo -e "\nToken: !$TOKEN"
+
+#--[ PIN alternative ]------------------------------------------------------------------------------
+
+HEXA=$(echo "$TOKEN" | base64 -d | xxd -p -u)
+HWORD=$(for ((i=0; i<=$(( ${#HEXA} / 6 )); i++)); do echo "$HEXA" | cut -c $(( i * 6 + 1 ))-$(( i * 6 + 6 )); done)
+XWORD=$(echo $HWORD | sed 's/ / ^ 0x/g')
+
+echo "PIN: ${TOKEN: -4}$(( 0x$XWORD ))"
